nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,326 @@
+                    y.data[NNCFGraph.NODE_NAME_ATTR] = x.node_name
+                    num_matches += 1
+            assert num_matches == 1
+
+    # We need to collect original bias values for biased nodes.
+    # It will be stored inside the `bised_node.data` dictionary.
+    # The original bias will be used to undo bias correction for
+    # the node if it is reverted to the original precision during
+    # the quantizer removing process.
+    nodes_with_bias = (node for node in nncf_graph.get_all_nodes() \
+                       if algo_backend.is_node_with_bias(node, nncf_graph))
+    for biased_node in nodes_with_bias:
+        biased_node.data['original_bias'] = algo_backend.get_bias_value(biased_node, nncf_graph, initial_model)
+
+    nodes_with_weight = (node for node in graph.get_all_nodes() \
+                         if algo_backend.is_node_with_weight(node))
+    for node_with_weight in nodes_with_weight:
+        node = nncf_graph.get_node_by_name(node_with_weight.node_name)
+        node.data['original_weight'] = algo_backend.get_weight_value(node_with_weight, graph, initial_model)
+
+    # Show the number of quantized operations.
+    num_of_quantized_ops = get_number_of_quantized_ops(nncf_graph,
+                                                       algo_backend.get_quantizer_metatypes(),
+                                                       algo_backend.get_quantizable_metatypes())
+    nncf_logger.info(f'The total number of quantized operations in the model: {num_of_quantized_ops}')
+
+    # Check whether it is possible to calculate the metric for one data item.
+    # pylint: disable=W0703
+    USE_METRIC = True
+    try:
+        _ = validation_fn(algo_backend.prepare_for_inference(initial_model),
+                          validation_dataset.get_data([0]))
+    except Exception:
+        USE_METRIC = False
+    nncf_logger.info(f'The {"original" if USE_METRIC else "NMSE"} metric will be used to rank quantizers')
+
+    # TODO(andrey-churkin): We need read dataset only once here to optimize execution time.
+    if USE_METRIC:
+        x_ref = get_metric_for_each_item(algo_backend.prepare_for_inference(initial_model),
+                                         validation_dataset,
+                                         validation_fn)
+
+        x_approx = get_metric_for_each_item(algo_backend.prepare_for_inference(quantized_model),
+                                            validation_dataset,
+                                            validation_fn)
+    else:
+        output_name = [x.node_name for x in nncf_graph.get_output_nodes()][0]
+        x_ref = get_logits_for_each_item(initial_model, validation_dataset, output_name)
+        x_approx = get_logits_for_each_item(quantized_model, validation_dataset, output_name)
+
+    groups_to_rank = find_groups_of_quantizers_to_rank(nncf_graph,
+                                                       algo_backend.get_quantizer_metatypes(),
```

Comment:
I am sorry but I don't understand what you mean.

Reply:
There are 3 calls of the `algo_backend.get_quantizer_metatypes()` in the algorithm.

Reply:
The time complexity of this method is O(1). I really don't see the benefit to create the variable
```
quantizer_metatypes = algo_backend.get_quantizer_metatypes()
```
and use it instead of `algo_backend.get_quantizer_metatypes()`.